elm-src/NixJSON.elm

```diff
@@ -0,0 +1,150 @@
+module NixJSON exposing (Gson, gnull, nixFromGson, gsonDecoder)
+
+import Dict exposing (Dict, empty, foldr, filter)
+import Json.Decode as Decode
+
+type Gson
+    = GString String
+    | GBool Bool
+    | GInt Int
+    | GFloat Float
+    | GList (List Gson)
+    | GDict (Dict String Gson)
+    | GNull
+
+gnull : Gson
+gnull = GNull
+
+nixFromGson : Gson -> String
+nixFromGson gson =
+    case gson of
+        GNull ->
+            "null"
+
+        GString x ->
+            nix_string x
+
+        GBool True ->
+            "true"
+
+        GBool False ->
+            "false"
+
+        GInt x ->
+            toString x
+
+        GFloat x ->
+            toString x
+
+        GList list ->
+            nix_list list
+
+        GDict dict ->
+            if ((Dict.member "_type" dict) && ((Dict.get "_type" dict) == Just (GString "literalExample")) && (Dict.member "text" dict)) then
```

Comment:
You could turn this from an `if/then` to a `case/of`:

```
case (Dict.get "_type" dict, Dict.get "text" dict) of
  (Just (GString "literalExample"), Just (GString txt) ->
    txt
  (Just (GString "literalExample"), _ ) -> 
    "Literal example failed to product text"
  _ ->
     elseBranchCodeHere
```

Reply:
Nice! Thank you!